### PR TITLE
First PR in a while it seems...

### PR DESCRIPTION
**Pull request by zzzbra**
Not really sure how I let that happened, could've swore I hadn't fallen behind at all...

### CODE REVIEW

File: zach_brady/class6/.ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+system
```
Comment (trivett) [inline zach_brady/class6/.ruby-version]: @zzzbra you probably don't need this any more now that you have your machine re-set up

File: zach_brady/class6/animals/lib/animal.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Animal
+    attr_accessor :num_legs
+
+    @@num_legs = 4
```
Comment (trivett) [inline zach_brady/class6/animals/lib/animal.rb]: A solution involving constants:

````
class Animal
    attr_accessor :num_legs

    NUM_LEGS = 4

    def to_s
        "This animal has #{ @num_legs } legs and responds to #{ @responds_to }"
    end
end



class Cat < Animal
    attr_accessor :eats_poop, :responds_to

    def initialize
        @eats_poop = false
        @responds_to = "pssshhhhhhhh pshh pshh pshh"
        @num_legs = NUM_LEGS
    end
end

c = Cat.new
# => #<Cat:0x007f93091a8378 @eats_poop=false, @num_legs=4, @responds_to="pssshhhhhhhh pshh pshh pshh">

````

File: zach_brady/mid-term/main.rb [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        puts "What is the animal's name? "
+        name = gets.chomp
+        puts "What is the animal's species? "
+        species = gets.chomp.downcase
+        toys = []
+        puts "Does #{name} have any toys? "
+        reply = gets.chomp.downcase
+        if reply == "yes"
+            puts "What is #{name}'s favorite toy? "
+            favorite_toy = gets.chomp
```
Comment (trivett) [inline zach_brady/mid-term/main.rb]: good thinkin

File: zach_brady/mid-term/main.rb [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    puts "  - (d) list all clients"
+    puts "  - (e) adoption"
+    puts "  - (f) return pet (you asshole.)"
+    puts "  - (q) quit"
+    puts
+
+    answer = gets.chomp.downcase
+    puts
+
+    case answer
```
Comment (trivett) [inline zach_brady/mid-term/main.rb]: one odd difference with JS is that case is actually slower than if/else, interestingly enough. Totally the opposite in JavaScript. I [found that out](http://midwire.github.io/blog/2011/08/26/ruby-performance-case-vs-if-elsif/) one time when I had a freaking massive if else in a JS project that too long enough to irritate a patient user. helped a tiny bit to use case.

File: zach_brady/mid-term/lib/shelter.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    end
+
+
+    def adopt(animal, client)
+        client.pets << animal
+    end
+
+    def return_pet(animal, client)
+        if client.pets.include? animal
+            client.pets.delete(animal)
```
Comment (trivett) [inline zach_brady/mid-term/lib/shelter.rb]: Does the animal get into the shelter or just get thrown into the Ruby garbage collector?

**Comment by trivett** [comment]
Nicely done. Any outstanding questions from the first three weeks that I might be able to help with? You got your environment all set and working as expected in your tmux sessions, right?

**Comment by zzzbra** [comment]
I do have one or two. I might pull you aside and clarify some of this unexpected behavior I observed re: the attr_accesor when my class has variables with defaults set

File: zach_brady/mid-term/main.rb (hunk 4 above)
Comment (zzzbra) [inline zach_brady/mid-term/main.rb]: Duly noted!

File: zach_brady/class6/animals/lib/animal.rb (hunk 2 above)
Comment (zzzbra) [inline zach_brady/class6/animals/lib/animal.rb]: this is helpful, but I will need to reconstruct some of the issues I was having to see if this recreates them.

**Pull request merged**